### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by srph**
Resolves https://github.com/vkeusebio/ug/issues/3.

- [x] Refactors task into a Job.
- [x] Uses `str_random` instead of custom string generator.
- [x] Ignores generated files and main env config.
- [x] Uses `Filesystem` instead of low-level functions like `fopen`, and etc.
- [x] Tests.

### CODE REVIEW

File: app/Jobs/GenerateStylesheetArchive.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+     *
+     * @param $temp Path to temp file
+     * @return void
+     */
+    protected function generate($temp)
+    {
+        // @refactor
+        // This code doesn't make any sense.
+        // We need to add a validation that at least a single stylesheet is required
+        $this->fs->append($temp, '');
```
Comment (srph) [inline app/Jobs/GenerateStylesheetArchive.php]: **Note**

**Comment by srph** [comment]
You can merge now if you want to. Don't forget to test it out. Good thing I remembered before falling asleep.

**Comment by vkeusebio** [comment]
@srph you did a lot, thank you so much! :smile:

**Comment by vkeusebio** [comment]
I'm having the following issues:

I see that you used session on the view.
```
Class 'Session' not found
```

after enabling the session middleware:
```
Fatal error: Arrays are not allowed in class constants in ug\app\Jobs\GenerateStylesheetArchive.php on line 31
```

**Comment by srph** [comment]
I didn't do anything to the view.

What PHP version are you using?

**Comment by srph** [comment]
Oh yea, don't forget to re-create your `.env` file again because I ignored it.

**Comment by vkeusebio** [comment]
I did recreated my .env file, I'm using PHP 5.6

**Comment by srph** [comment]
`composer dumpautolaod && composer update`

How about that?

**Comment by vkeusebio** [comment]
Oh wait, I'm actually using PHP 5.5.27. Chat? sure thing.

**Comment by srph** [comment]
I'm using 5.6

**Comment by vkeusebio** [comment]
That's weird. Where can we talk in real-time?

**Comment by srph** [comment]
Gitter, Facebook, Twitter.

**Comment by vkeusebio** [comment]
Fixed it.

```php
protected $stylesheets = array(
    'Base',
    ...
    'Utility'
);

...

protected function generate($temp)
{
    ....
    foreach($this->stylesheets as $stylesheet) {
        ...
    }
}
```

Could you replace the said lines on `GenerateStylesheetArchive.php` like the ones above?

**Comment by srph** [comment]
Hmm, I like my constants in here, but okay, thanks! Lemme fix it in a sec.

**Comment by vkeusebio** [comment]
Thank you so much for all the help @srph , I really appreciate it :+1:

**Comment by srph** [comment]
No prob, learn a few stuff along the way.

My bad for the slow replies. Stressed out these days, juggling lots of stuff sucks.

**Pull request merged**